### PR TITLE
Added tag container support

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Adding tag container support so we can be explicit about the container to run in

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,32 +22,32 @@ jobs:
...
-            { "name": "sles-15", "arch": "amd64" },
-            { "name": "ubuntu-20.04", "arch": "amd64" },
-            { "name": "ubuntu-22.04", "arch": "amd64" }
+            { "name": "amazonlinux-2", "arch": "amd64", "tag": "latest" },
+            { "name": "centos-7", "arch": "amd64", "tag": "latest" },
+            { "name": "centos-8", "arch": "amd64", "tag": "latest" },
+            { "name": "debian-10", "arch": "amd64", "tag": "latest" },
+            { "name": "debian-11", "arch": "amd64", "tag": "latest" },
+            { "name": "debian-12", "arch": "amd64", "tag": "latest" },
+            { "name": "mariner-2", "arch": "amd64", "tag": "latest" },
```
Comment (MariusNi) [inline .github/workflows/ci.yml:31]: >{ "name": "mariner-2", "arch": "amd64", "tag": "latest" }, [](http://example.com/codeflow?start=12&length=58)

Rename to Azure Linux 2 and move to the top of the list, pls

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,32 +22,32 @@ jobs:
...
-            { "name": "mariner-2", "arch": "amd64" },
-            { "name": "oraclelinux-7", "arch": "amd64" },
-            { "name": "oraclelinux-8", "arch": "amd64" },
-            { "name": "rhel-7", "arch": "amd64" },
-            { "name": "rhel-8", "arch": "amd64" },
-            { "name": "rhel-9", "arch": "amd64" },
-            { "name": "sles-15", "arch": "amd64" },
-            { "name": "ubuntu-20.04", "arch": "amd64" },
-            { "name": "ubuntu-22.04", "arch": "amd64" }
+            { "name": "amazonlinux-2", "arch": "amd64", "tag": "latest" },
```
Comment (MariusNi) [inline .github/workflows/ci.yml:25]: >{ "name": "amazonlinux-2", "arch": "amd64", "tag": "latest" }, [](http://example.com/codeflow?start=12&length=62)

AlmaLinux 9 missing?

File: .github/workflows/insiders-fast.yml [hunk 3]
```diff
@@ -33,25 +33,25 @@ jobs:
...
+            { os: debian, version: 10, arch: amd64, dist: buster, package-type: DEB, repo: microsoft-debian-buster-prod-apt, container-tag: latest },
+            { os: debian, version: 10, arch: arm64, dist: buster, package-type: DEB, repo: microsoft-debian-buster-prod-apt, container-tag: sha-c689eee },
+            { os: debian, version: 11, arch: amd64, dist: bullseye, package-type: DEB, repo: microsoft-debian-bullseye-prod-apt, container-tag: latest },
+            { os: debian, version: 11, arch: arm64, dist: bullseye, package-type: DEB, repo: microsoft-debian-bullseye-prod-apt, container-tag: sha-c689eee },
+            { os: debian, version: 12, arch: amd64, dist: bookworm, package-type: DEB, repo: microsoft-debian-bookworm-prod-apt, container-tag: latest },
+            { os: debian, version: 12, arch: arm64, dist: bookworm, package-type: DEB, repo: microsoft-debian-bookworm-prod-apt, container-tag: sha-db3d4c8 },
+            { os: mariner, version: 2, arch: amd64, dist: azurelinux, package-type: RPM, repo: cbl-mariner-2.0-preview-Microsoft-x86_64-yum, container-tag: latest },
```
Comment (MariusNi) [inline .github/workflows/insiders-fast.yml:45]: >{ os: mariner [](http://example.com/codeflow?start=12&length=13)

Rename to Azure Linux 2 and move to top of list, please.

File: .github/workflows/insiders-fast.yml [hunk 4]
```diff
@@ -33,25 +33,25 @@ jobs:
...
-            { os: rhel, version: 9, arch: amd64, dist: rhel9, package-type: RPM, repo: microsoft-rhel9.0-insiders-fast-prod-yum },
-            { os: rockylinux, version: 9, arch: amd64, dist: rockylinux9, package-type: RPM, repo: microsoft-el9-insiders-fast-prod-yum },
-            { os: sles, version: 15, arch: amd64, dist: sles15, package-type: RPM, repo: microsoft-sles15-insiders-fast-prod-yum },
-            { os: ubuntu, version: 20.04, arch: amd64, dist: focal, package-type: DEB, repo: microsoft-ubuntu-focal-prod-apt },
-            { os: ubuntu, version: 20.04, arch: arm64, dist: focal, package-type: DEB, repo: microsoft-ubuntu-focal-prod-apt },
-            { os: ubuntu, version: 22.04, arch: amd64, dist: jammy, package-type: DEB, repo: microsoft-ubuntu-jammy-prod-apt },
-            { os: ubuntu, version: 22.04, arch: arm64, dist: jammy, package-type: DEB, repo: microsoft-ubuntu-jammy-prod-apt },
+            { os: amazonlinux, version: 2, arch: amd64, dist: amazonlinux2, package-type: RPM, repo: microsoft-amazonlinux2-insiders-fast-prod-yum, container-tag: latest },
```
Comment (MariusNi) [inline .github/workflows/insiders-fast.yml:36]: >{ os: amazonlinux,  [](http://example.com/codeflow?start=12&length=19)

AlmaLinux 9?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (AhmedBM) [reply .github/workflows/ci.yml:31]: Thats part of a larger change, those targets must match the Dockerfiles, which is set to mariner sadly. Ill have a follow-up PR for when native arm64 targets come with this fixed.

File: .github/workflows/insiders-fast.yml (hunk 3 above)
Comment (AhmedBM) [reply .github/workflows/insiders-fast.yml:45]: same as above, must stay the same as the container reference.

File: .github/workflows/insiders-fast.yml (hunk 4 above)
Comment (AhmedBM) [reply .github/workflows/insiders-fast.yml:36]: added

Comment (MariusNi) [reply .github/workflows/insiders-fast.yml:36]: Where? Here is still only Rocky in list, no Alma

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**